homework-submission/todoapp/src/Components/TodoElement.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+class TodoElement extends React.Component {
+  state = {
+    isEdit: false
+
+  }
+  handleEditButton = () => {
+    this.setState({
+      isEdit: true
+    })
+  }
+  handleCancelButton = () => {
+    this.setState({
+      isEdit: false
+    })
+  }
+  handleUpdate = (e) => {
+    e.preventDefault()
+    let oldTask = this.props.description
+    let newTask = this.editField.value
+    let newDate = this.secondEditField.value
+    this.props.updateTodo(oldTask, newTask, newDate)
+    this.setState({
+      isEdit: false
+
+    })
+
+  }
+
+  render() {
+    let TodoView
+    if (!this.state.isEdit) {
+      TodoView = (
```

Comment:
### Extra Challenge 💪👌
Think to make `TodoView` a bit simpler.

_May be a new component?_

Reply:
I have made some changes but i could not fix all issues. Needs some help

Reply:
Which part? 🙂